contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol

```diff
@@ -388,31 +392,109 @@ contract FillQuoteTransformer is
+                    outputTokenAmount,
+                    makerToken)
+            returns (FillOrderResults memory fillResults)
+            {
+                results = fillResults;
+            } catch (bytes memory) {
+                // Swallow failures, leaving all results as zero.
+            }
+        } else {
+            // Track changes in the maker token balance.
+            uint256 initialMakerTokenBalance = makerToken.balanceOf(address(this));
+            try
+                exchange.fillOrder
+                    {value: protocolFee}
+                    (order, takerAssetFillAmount, signature)
+                returns (IExchange.FillResults memory fillResults)
+            {
+                // Update maker quantity based on changes in token balances.
+                results.makerTokenBoughtAmount = makerToken.balanceOf(address(this))
+                    .safeSub(initialMakerTokenBalance);
+                // We can trust the other fill result quantities.
+                results.protocolFeePaid = fillResults.protocolFeePaid;
+                results.takerTokenSoldAmount = fillResults.takerAssetFilledAmount;
+                // If the taker fee is payable in the taker asset, include the
+                // taker fee in the total amount sold.
+                if (isTakerFeeInTakerToken) {
+                    results.takerTokenSoldAmount =
+                        results.takerTokenSoldAmount.safeAdd(fillResults.takerFeePaid);
+                }
+            } catch (bytes memory) {
+                // Swallow failures, leaving all results as zero.
             }
-        } catch (bytes memory) {
-            // Swallow failures, leaving all results as zero.
         }
     }
 
+    /// @dev Attempt to fill a ERC20 Bridge order. If the fill reverts,
+    ///      or the amount filled was not sufficient this reverts.
+    /// @param order The bridge order to fill.
+    /// @param inputTokenAmount How much taker asset to fill.
+    /// @param outputTokenAmount How much maker asset to receive.
+    /// @param makerToken The maker token.
+    function fillBridgeOrder(
+        IExchange.Order calldata order,
+        uint256 inputTokenAmount,
+        uint256 outputTokenAmount,
+        IERC20TokenV06 makerToken
+    )
+        external
+        returns (FillOrderResults memory results)
+    {
+        require(msg.sender == address(this), "SENDER_NOT_AUTHORIZED");
+        // Track changes in the maker token balance.
+        uint256 initialMakerTokenBalance = makerToken.balanceOf(address(this));
+        (
+            address tokenAddress,
+            address bridgeAddress,
+            bytes memory bridgeData
+        ) = abi.decode(
+            LibBytesV06.sliceDestructive(
+                order.makerAssetData,
+                4,
+                order.makerAssetData.length),
+            (address, address, bytes)
+        );
+        require(bridgeAddress != address(this), "INVALID_BRIDGE_ADDRESS");
```

Comment:
I'm open to better revert reasons here, but ultimately these are just caught and ignored, not surfaced to the user.